snippets/core/system-text-json/csharp/RoundtripToFileAsync.cs

```diff
@@ -29,6 +30,24 @@ public static async Task RunAsync()
             }
             // </SnippetDeserialize>
             weatherForecast.DisplayPropertyValues();
+
+            using (FileStream fs = File.Create(fileNameUtf8))
+            {
+                using (StreamWriter writer = new StreamWriter(fs, Encoding.UTF8))
```

Comment:
What's the point of this `writer`?

Reply:
This code isn't displayed in the doc. My intent was to provide a snippet showing how to write a file encoded as UTF-8 and to get this code validated or corrected before showing it in the doc.

Reply:
You can write data directly to the `FileStream` using the JsonSerializer/Utf8JsonWriter APIs. That's how you emit UTF-8 encoded data. You don't need `StreamWriter`.

Reply:
When I change the code to use JsonSerializer directly, the exception goes away. It only happens with the file created by the StreamWriter specifying UTF-8 encoding.  I'm fixing related parts of the doc now.

Reply:
> the exception goes away. It only happens with the file created by the StreamWriter specifying UTF-8 encoding

Which exception?